diff -Naupr couchbase-4.2.4.orig/src/wrapper/connection_handle.cxx couchbase-4.2.4/src/wrapper/connection_handle.cxx
--- couchbase-4.2.4.orig/src/wrapper/connection_handle.cxx	2024-10-23 02:26:12.000000000 +0200
+++ couchbase-4.2.4/src/wrapper/connection_handle.cxx	2024-11-05 15:31:12.436652043 +0100
@@ -2096,6 +2096,9 @@ connection_handle::document_upsert_multi
   if (auto e = cb_set_durability(opts, options); e.ec) {
     return e;
   }
+  if (auto e = cb_set_expiry(opts, options); e.ec) {
+    return e;
+  }
   if (auto e = cb_set_preserve_expiry(opts, options); e.ec) {
     return e;
   }
