diff --git a/exporter/exporter.go b/exporter/exporter.go
index cae2825..8f77a96 100644
--- a/exporter/exporter.go
+++ b/exporter/exporter.go
@@ -42,15 +42,15 @@ type Exporter struct {
 func NewExporter(uri url.URL, insecure bool, user, password string) *Exporter {
 	q := uri.Query()
 	metricsURI := uri
-	q.Set("query", "select metric, value from system.metrics")
+	q.Set("query", "select metric, toInt64(value) from system.metrics")
 	metricsURI.RawQuery = q.Encode()
 
 	asyncMetricsURI := uri
-	q.Set("query", "select metric, value from system.asynchronous_metrics")
+	q.Set("query", "select metric, toInt64(value) from system.asynchronous_metrics")
 	asyncMetricsURI.RawQuery = q.Encode()
 
 	eventsURI := uri
-	q.Set("query", "select event, value from system.events")
+	q.Set("query", "select event, toInt64(value) from system.events")
 	eventsURI.RawQuery = q.Encode()
 
 	partsURI := uri
