diff -Naupr php-client-0.2.0.orig/Cargo.toml php-client-0.2.0/Cargo.toml
--- php-client-0.2.0.orig/Cargo.toml	2023-10-25 18:21:45.000000000 +0000
+++ php-client-0.2.0/Cargo.toml	2023-11-15 22:17:02.443285946 +0000
@@ -10,7 +10,7 @@ crate-type = ["cdylib"]
 ordered-float = { version = "3", default-features = false }
 aerospike-sync = { git = "https://github.com/aerospike/aerospike-client-rust.git", branch = "php-rs", package = "aerospike-sync" }
 aerospike-core = { git = "https://github.com/aerospike/aerospike-client-rust.git", branch = "php-rs", package = "aerospike-core" }
-ext-php-rs = "0.10.1"
+ext-php-rs = "=0.10.1"
 colored = "2.0.0"
 hex = "0.4"
 log = "0.4"
