diff -Naupr device-detection-cxx-4.2.3.orig/src/hash/hash.c device-detection-cxx-4.2.3/src/hash/hash.c
--- device-detection-cxx-4.2.3/src/hash/hash.c	2021-05-10 11:31:14.000000000 +0200
+++ device-detection-cxx-4.2.3/src/hash/hash.c	2021-05-27 20:19:43.976629178 +0200
@@ -1344,7 +1344,8 @@ static void initGetEvidencePropertyRelat
 	DataReset(&propertyItem.data);
 	DataReset(&nameItem.data);
 	int propertiesCount = CollectionGetCount(dataSet->properties);
-	for (int propertyIndex = 0; 
+	int propertyIndex;
+	for (propertyIndex = 0; 
 		propertyIndex < propertiesCount && EXCEPTION_OKAY; 
 		propertyIndex++) {
 		property = PropertyGet(
@@ -2259,7 +2260,8 @@ void fiftyoneDegreesResultsHashFromEvide
 		// situations where a User-Agent that is provided by the calling
 		// application can be used in preference to the one associated with the
 		// calling device.
-		for (int i = 0;
+		int i;
+		for (i = 0;
 			i < FIFTYONE_DEGREES_ORDER_OF_PRECEDENCE_SIZE &&
 			results->count == 0;
 			i++) {
@@ -2393,7 +2395,8 @@ createPseudoEvidenceKeyValueArray(
 				(void*)Malloc(
 					pseudoEvidence->capacity * maxUaLength);
 			if (evidenceMem != NULL) {
-				for (uint32_t i = 0; i < pseudoEvidence->capacity; i++) {
+				uint32_t i;
+				for (i = 0; i < pseudoEvidence->capacity; i++) {
 					pseudoEvidence->items[i].field = NULL;
 					pseudoEvidence->items[i].originalValue =
 						(void*)((char*)evidenceMem + i * maxUaLength);
