diff -Naupr vim-puppet-master.orig/after/ftplugin/puppet.vim vim-puppet-master/after/ftplugin/puppet.vim
--- vim-puppet-master.orig/after/ftplugin/puppet.vim	2024-09-15 22:30:38.000000000 +0200
+++ vim-puppet-master/after/ftplugin/puppet.vim	2025-01-23 10:17:07.139260703 +0100
@@ -1,5 +1,5 @@
 if !exists('g:puppet_align_hashes')
-    let g:puppet_align_hashes = 1
+    let g:puppet_align_hashes = 0
 endif
 
 if g:puppet_align_hashes
